src/common/storage/project/cache.js

```diff
@@ -23,12 +27,28 @@ define(['common/util/assert', 'common/storage/constants'], function (ASSERT, CON
         this.queuedPersists = {};
 
         function cacheInsert(key, obj) {
+            var patch;
             ASSERT(obj[CONSTANTS.MONGO_ID] === key);
             logger.debug('cacheInsert', key);
 
             //deepFreeze(obj);
             if (!cache[key]) {
-                cache[key] = obj;
+                if(obj.type === 'patch'){
+                    if(cache[obj.base] || backup[obj.base]){
```

Comment:
Would it make sense to call `getFromCache` here?
Or the only time a patch object is inserted is when we receive it from another user (so it couldn't be in the makeCommit queue..)?